doc/ubertooth.7
.TH UBERTOOTH 7 "March 2017" "Project Ubertooth" "Project Ubertooth Overview"
.SH NAME
.PP
.BR ubertooth (7) 
\- Project Ubertooth
.SH OVERVIEW
.PP
Ubertooth is a platform for Bluetooth experimentation. It is able to
sniff Bluetooth Smart (BLE), discover undiscoverable classic Bluetooth
devices, and perform rudimentary sniffing of classic Bluetooth devices.
.PP
Ubertooth is \fBnot\fP a Bluetooth dongle and it will not show up as one
when running \fB\fChciconfig\fR\&.
.SH COMMAND LINE TOOLS
.PP
The following command line tools included in this package are the
primary means of using Ubertooth. Refer to their manual pages for
example usage and comprehensive documentation.
.PP
Useful commands:
.RS
.IP \(bu 2
.BR ubertooth-btle (1) 
: BLE sniffing and other fun
.IP \(bu 2
.BR ubertooth-rx (1) 
: Classic Bluetooth device discovery and rudimentary sniffing
.IP \(bu 2
.BR ubertooth-scan (1) 
: Active scanning of undiscoverable devices
.IP \(bu 2
.BR ubertooth-afh (1) 
: Detecting the AFH map of a Classic Bluetooth piconet
.IP \(bu 2
.BR ubertooth-ego (1) 
: Yuneec E\-Go electric skateboard sniffing
.RE
.PP
Utility commands:
.RS
.IP \(bu 2
.BR ubertooth-dfu (1) 
: Firmware update tool
.IP \(bu 2
.BR ubertooth-dump (1) 
: Dumping raw RF symbols to disk
.IP \(bu 2
.BR ubertooth-util (1) 
: "Everything else"
.RE
.PP
Less useful commands:
.RS
.IP \(bu 2
.BR ubertooth-debug (1) 
: Peeking and poking registers on the CC2400
.IP \(bu 2
.BR ubertooth-specan (1) 
: Raw RSSI values used by graphical specan
.IP \(bu 2
.BR ubertooth-tx (1) 
: Retransmitting raw RF symbols dumped with 
.BR ubertooth-dump (1)
.RE
.SH SUPPORT
.PP
Ubertooth is an open source project maintained primarily by volunteers.
Please keep that in mind when seeking support.
.PP
If you are having issues with a tool or believe you have found a bug,
please file an issue on the project GitHub:
\[la]https://github.com/greatscottgadgets/ubertooth\[ra]
.PP
For interactive support, join the \fB\fC#ubertooth\fR IRC channel on Freenode.
.SH COPYRIGHT
.PP
The code and documentation for Project Ubertooth are Copyright 2010\-2017
and are available under the terms of the GPLv2. Refer to \fB\fCCOPYING\fR for
details. Ubertooth is a registered trademark of Great Scott Gadgets.
.SH AUTHOR
.PP
This manual page was written by Mike Ryan.


changelog
ubertooth (2017-03-R2) unstable; urgency=low
  * USB API versioning - We have added USB API versioning to improve the
    experience of failures due to mismatched firmware and host code. You may
    find that Ubertooth tools complain that your firmware is out of date.
    Updating firmware is simple and reliable, please follow the instructions
    on the wiki. A release archive with precompiled firmware is available.
  * Basic Rate PCAP output - the pcap file produced by the Ubertooth tools
    should all be read by the latest Wireshark versions without the need to
    install plugins.
  * Kismet plugins - Kismet is currently under heavy development and it has
    not been possible to continue tracking the master branch. The legacy
    Kismet plugins should continue to work, but we will wait to track git
    master until development has stabalised.
  
 -- Dominic Spill <dominicgs@gmail.com>  01 Mar 2017 10:39:00 -0700   
 
 ubertooth (2015-10-R1) unstable; urgency=low
  * This is a minor bugfix and update release
  * Update ubertooth-scan to provide more consistent output
  * It is now possible to build receive only firmware
  * Host code versioning is more reliable
  * Python tools now support Python 3. Thanks Hannes!

  
 -- Dominic Spill <dominicgs@gmail.com>  12 Oct 2015 03:10:00 -0700   
 
 ubertooth (2015-09-R2) unstable; urgency=low
  * Basic Rate Bluetooth Improvements
      * Stability has improved in performance constrained environments
      * The "uh oh, buffer not emptied" issue no longer causes the tools to loop
        indefinitely
      * AFH support has greatly improved, UAP and CLK values are now found more
        quickly when AFH is used
  * Basic Rate PCAP support
      * ubertooth-rx and ubertooth-follow support a new PCAP file format
      * The new PCAP format is also supported with a new Wireshark plugin: btbredr
      * The PCAP format is still in the early stages, but we hope to have the plugin
        included in Wireshark later this year
      * The older btbb format is still produced by the Kismet plugins, they will
        be migrated over to the new format in the next release
  * Bluetooth Smart (BLE) Improvements
      * Improved packet reception
      * Promiscuous sniffing mode is more reliable
      * Promiscuous sniffing reports the recovered parameters to the host
  * Skateboard Hacking
      * Support for interfering with connections of BLE enabled skateboards
      * ubertooth-ego targets Yuneec E-go (not Bluetooth)
      * Always hack safely and responsibly!

  
 -- Dominic Spill <dominicgs@gmail.com>  05 Sep 2015 05:30:00 -0700   

ubertooth (2014-02-R2) unstable; urgency=low
  * Bluetooth Smart (Low Energy) Support
      * Promiscuous and follow modes
      * Pcap format packet logging
      * Pairing / encryption support when paired with crackle
      * Credit for BLE features goes to Mike Ryan
  * Unified host tool for monitoring Basic Rate
      * ubertooth-rx replaces -lap, -uap, -hop tools
      * Once UAP is discovered, ubertooth-rx automatically tries to find clock
        values and begin hopping
      * Thanks to Will Code for working on this
  * Survey tool - ubertooth-scan
      * Combining both Ubertooth and a standard Bluetooth dongle
      * Ubertooth scans for non-discoverable master devices
      * Dongle probes devices for piconet information and features
  * Cmake now used for the build system
      * Improves support for non-Linux operating systems
      * More sensible handling of dependencies
      * New build instructions
  * Packaging (Experimental)
      * Early stage support for packaging systems
      * libbtbb in Homebrew repository, Ubertooth coming soon
      * MacPorts availability is under test
      * Release already available in Pentoo
  * GitHub migration
      * libbtbb, Ubertooth and gr-bluetooth all hosted on GitHub
      * Allows for more open development and collaboration model
      * Already seeing an increase in issue reporting and pull requests


 -- Dominic Spill <dominicgs@gmail.com>  20 Feb 2014 06:06:00 -0700   


host/packaging/ubertooth.rb
require "formula"

class Ubertooth < Formula
  homepage "https://github.com/greatscottgadgets/ubertooth"
  url "https://github.com/greatscottgadgets/ubertooth/archive/2014-02-R2.tar.gz"
  sha1 "773eab8b6440dcf7fd6bcc83d9e365c3239dd5f3"
  version "2014-02-R2"

  head "https://github.com/greatscottgadgets/ubertooth.git"

  resource "pyusb" do
    url "https://pypi.python.org/packages/source/p/pyusb/pyusb-1.0.0b1.tar.gz"
    sha1 "f0ca8bdfbb59645ea73976f9a3cb2b0701667148"
  end

  depends_on "cmake" => :build
  depends_on :python
  depends_on "libbtbb"

  def install
    ENV.prepend_create_path "PYTHONPATH", libexec+"lib/python2.7/site-packages"
    resource("pyusb").stage { system "python", "setup.py", "install", "--prefix=#{libexec}" }

    mkdir "host/build" do
      system "cmake", "..", *std_cmake_args
      system "make", "install"
    end
  end
end
